src/seamless_communication/models/inference/translator.py

```diff
@@ -74,10 +73,22 @@ def __init__(
             pad_idx=self.text_tokenizer.vocab_info.pad_idx, pad_to_multiple=2
         )
         # Load the vocoder.
-        self.vocoder: Vocoder = load_vocoder_model(vocoder_name_or_card, device=device)
-        self.vocoder.eval()
+        self.vocoder: Vocoder = self.load_model_for_inference(
+            load_vocoder_model, vocoder_name_or_card, device, torch.float32
+        )
         self.sr = sample_rate
 
+    @staticmethod
+    def load_model_for_inference(
+        load_model_fn: Any,
```

Comment:
nit: being pedantic here. Ideally this should be `Callable[[...], torch.nn.Module]`.

Reply:
How about we go with `Callable[[Union[str, AssetCard], Device, DataType], nn.Module]`?